SimG4Core/CustomPhysics/src/G4muDarkBremsstrahlungModel.cc

```diff
@@ -0,0 +1,454 @@
+  gsl_integration_qags(&F, tmin, tmax, 0, 1e-7, 1000, w, &result, &error);
+
+  G4double ChiRes = result;
+  gsl_integration_workspace_free(w);
+
+  //Integrate over x. Can use log approximation instead, which falls off at high A' mass.
+  gsl_integration_workspace* dxspace = gsl_integration_workspace_alloc(1000);
+  gsl_function G;
+  G.function = &DsigmaDx;
+  G.params = &alpha;
+  G4double xmin = 0;
+  G4double xmax = 1;
+  if ((Mmu / E0) > (MA / E0))
+    xmax = 1 - Mmu / E0;
+  else
+    xmax = 1 - MA / E0;
+  G4double res, err;
+
+  gsl_integration_qags(&G, xmin, xmax, 0, 1e-7, 1000, dxspace, &res, &err);
+
+  G4double DsDx = res;
+  gsl_integration_workspace_free(dxspace);
+
+  G4double GeVtoPb = 3.894E08;
+  G4double alphaEW = 1.0 / 137.0;
+  G4double epsilBench = 1;
+
+  cross = GeVtoPb * 4. * alphaEW * alphaEW * alphaEW * epsilBench * epsilBench * ChiRes * DsDx * CLHEP::picobarn;
+  if (cross < 0.) {
+    cross = 0.;
+  }
+  E0 = E0 * CLHEP::GeV;
+
+  cross = cross * cxBias;
+  return cross;
+}
+
+G4DataVector* G4muDarkBremsstrahlungModel::ComputePartialSumSigma(const G4Material* material,
+                                                                  G4double kineticEnergy,
+                                                                  G4double cut)
+
+// Build the table of cross section per element.
+//The table is built for MATERIALS.
+// This table is used by DoIt to select randomly an element in the material.
+{
+  G4int nElements = material->GetNumberOfElements();
+  const G4ElementVector* theElementVector = material->GetElementVector();
+  const G4double* theAtomNumDensityVector = material->GetAtomicNumDensityVector();
+  G4DataVector* dv = new G4DataVector();
+  G4double cross = 0.0;
+
+  for (G4int i = 0; i < nElements; i++) {
+    cross += theAtomNumDensityVector[i] *
+             ComputeCrossSectionPerAtom(
+                 particle, kineticEnergy, (*theElementVector)[i]->GetZ(), (*theElementVector)[i]->GetA(), cut);
+    dv->push_back(cross);
+  }
+  return dv;
+}
+
+void G4muDarkBremsstrahlungModel::SampleSecondaries(std::vector<G4DynamicParticle*>* vdp,
+                                                    const G4MaterialCutsCouple* couple,
+                                                    const G4DynamicParticle* dp,
+                                                    G4double tmin,
+                                                    G4double maxEnergy)
+//Simulates the emission of a dark photon + electron. Gets an energy fraction and Pt from madgraph files. Scales the energy so that the fraction of kinectic energy is constant, keeps the Pt constant. If the Pt is larger than the new energy, that event is skipped, and a new one is taken from the file.
+{
+  //Deactivate the process after one dark brem. Needs to be reactivated in the end of event action. If this is in the stepping action instead, more than one brem can occur within each step.
+  G4bool state = false;
+  G4String pname = "muDBrem";
```

Comment:
@revering , it is a minor thing but better use G4 mass (avoiding problems in corner cases):

double muMass = dp->GetDefinition()->GetPDGMass(); // this is exact muon mass in internal units (MeV)